baseline/pytorch/lm/train.py

```diff
@@ -46,15 +47,16 @@ def _get_pytorch_model(self):
         return self.model.module if self.gpus > 1 else self.model
 
     @staticmethod
-    def _get_dims(batch_dict):
+    def _get_dims(loader):
+        batch_dict = loader.dataset[0]
         return batch_dict['y'].shape
 
     @staticmethod
     def _num_toks(batch_dict):
-        return np.prod(LanguageModelTrainerPyTorch._get_dims(batch_dict))
+        return np.prod(batch_dict['y'].shape)
```

Comment:
I see that we don't use the `_get_dims` anymore because this `_num_toks` need to be dynamic (needs to report for each batch because it can be different) but that raises the question of if we can remove the `_get_dims` function.

Also although we always uses dense LM batches would it make sense to calculate the number of tokens based on the lengths in case we ever need a LM that has padding (like each example if a single sentence)

Reply:
We currently use `_get_dims` in cases where we are doing truncated backprop, and its a bit unfortunate that we get the batch size this way as it makes the code clunky and not very robust -- it ends up assuming that we have a map-style dataset which we dont want to assume probably moving forward.